mirrord/kube/src/api/runtime.rs

```diff
@@ -101,6 +107,52 @@ impl RuntimeData {
             container_name,
         })
     }
+
+    pub async fn check_node(&self, client: &kube::Client) -> Result<()> {
+        let node_api: Api<Node> = Api::all(client.clone());
+        let pod_api: Api<Pod> = Api::all(client.clone());
+
+        let node = node_api.get(&self.node_name).await?;
+
+        let allocatable = node
+            .status
+            .and_then(|status| status.allocatable)
+            .ok_or_else(|| KubeApiError::NodeBadAllocatable(self.node_name.clone()))?;
+
+        let allowed: u32 = allocatable
+            .get("pods")
+            .and_then(|Quantity(quantity)| quantity.parse().ok())
+            .ok_or_else(|| KubeApiError::NodeBadAllocatable(self.node_name.clone()))?;
+
+        let mut pod_count = 0;
+        let mut list_params = ListParams {
+            field_selector: Some(format!("spec.nodeName={}", self.node_name)),
+            ..Default::default()
+        };
+
+        loop {
+            let pods_on_node = pod_api.list(&list_params).await?;
```

Comment:
I believe this might fail and in this case should just issue a warning and continue.